### PR TITLE
Accelerate some adjustment for mixed precision

### PR DESCRIPTION
**Pull request by BenjaminBossan**
This PR makes 3 adjustments. Checking the individual commits may help with reviewing.

1. [Use accelerator.autocast when computing loss](https://github.com/skorch-dev/skorch/commit/c6b94cce548c7846cd13e35d5a32de66ebfc93da)

According to the accelerate docs, loss computation should be performed within the `accelerator.autocast` context manager:

https://huggingface.co/docs/accelerate/v0.21.0/en/quicktour#mixed-precision-training

I tested if this makes a difference by running the [this notebook](https://nbviewer.org/github/skorch-dev/skorch/blob/master/notebooks/Hugging_Face_Finetuning.ipynb) with fp16 precision.

I found no difference at all: The runtime was practically the same and the losses were identical. Still, I think it's better to have this than not, as it is recommended by the accelerate docs.

2. [Update LR scheduler callback to work w/ accelerate](https://github.com/skorch-dev/skorch/commit/ba5f15323920b331c99da476bdff37c326c1cb37)

According to the same docs, the LR scheduler step should sometimes be skipped when using mixed precision training because accelerate may skip update steps internally. Therefore, I updated the LR scheduler callback to check if the net has an accelerator and if it does, to check if a step is necessary.

This is actually quite hard to test because the necessity of stepping depends on accelerate's internal logic, which we don't want to test, and which might change in the future. Therefore, the added test just runs training with accelerate, mixed precision, and some LR schedulers, verifying that there is no error.

When running these tests + the normal LR scheduler tests locally on a machine that supports fp16, I get 100% line coverage of `lr_scheduler.py`. I think this is good enough.

3. [Non-functional clean ups related to LR schedulers](https://github.com/skorch-dev/skorch/commit/0cf1dc30eab8a840898626dcace5e3652e25a85b)

While working on the fixes in this PR, I also cleaned up some LR scheduler code. These clean ups are non-functional.

1. We imported `CyclicLR` as `TorchCyclicLR`. I'm not sure why but it is somehow related to very old PyTorch versions we no longer support, so I removed this.
2. Fixed some indentations in conditional checks to improve readability.

### CODE REVIEW

File: skorch/callbacks/lr_scheduler.py [hunk 1]
```diff
@@ -142,6 +137,32 @@ def on_train_begin(self, net, **kwargs):
...
+            if score is None:
+                lr_scheduler.step()
+            else:
+                lr_scheduler.step(score)
+        else:
+            if not accelerator_maybe.optimizer_step_was_skipped:
+                if score is None:
+                    lr_scheduler.step()
+                else:
+                    lr_scheduler.step(score)
```
[the remark this answers is not in the document]
Comment (BenjaminBossan) [reply skorch/callbacks/lr_scheduler.py:164]: Hmm, should it not be `if not accelerator_maybe or not accelerator_maybe.optimizer_step_was_skipped:`?

Given that at least I'm struggling with the logic, maybe just leave the more verbose implementation as is?

File: skorch/callbacks/lr_scheduler.py [hunk 2]
```diff
@@ -142,6 +137,32 @@ def on_train_begin(self, net, **kwargs):
...
+
+        1. If the lr scheduler is ReduceLROnPlateau, we need to pass the score.
+        2. If the net is uses AccelerateMixin, stepping has to be skipped in
+           certain conditions.
+
+        For more info on the latter, see:
+        https://huggingface.co/docs/accelerate/quicktour#mixed-precision-training
+
+        """
+        accelerator_maybe = getattr(net, 'accelerator', None)
```
[the remark this answers is not in the document]
Comment (BenjaminBossan) [reply skorch/callbacks/lr_scheduler.py:153]: Yes, I wanted (and succeeded so far) to avoid it, but here, I think this might be the best solution. 

Alternatively, we could have a method on the net that returns `False` by default and proxies `accelerator_maybe.optimizer_step_was_skipped` on `AccelerateMixin`, do you think that's better?

File: skorch/callbacks/lr_scheduler.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (thomasjpfan) [reply skorch/callbacks/lr_scheduler.py:164]: Hmm how about:

```python
accelerator_step_skipped = accelerator_maybe and accelerator_maybe.optimizer_step_was_skipped
if not accelerator_step_skipped:
    if score is None:
        lr_scheduler.step()
    else:
        lr_scheduler.step(score)
```

File: skorch/callbacks/lr_scheduler.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomasjpfan) [reply skorch/callbacks/lr_scheduler.py:153]: > do you think that's better?

I do not think it's better. I'm okay with keeping the current version.

File: skorch/callbacks/lr_scheduler.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenjaminBossan) [reply skorch/callbacks/lr_scheduler.py]: Yes, I like that. The code has been adapted, with a slight change to use early return if `accelerator_step_skipped`.

**Review by thomasjpfan — APPROVED** [review]
